lib/Importer/ONNXModelLoader.cpp

```diff
@@ -385,28 +403,67 @@ llvm::Error ONNXModelLoader::loadSlice(const ONNX_NAMESPACE::NodeProto &op,
   auto dims = data.dims();
   auto numDims = dims.size();
 
-  // Attributes 'starts' and 'ends' are mandatory and must be consistent.
-  RETURN_ERR_IF_NOT(dict.count("starts"),
-                    "Slice: attribute 'starts' is mandatory.");
-  RETURN_ERR_IF_NOT(dict.count("ends"),
-                    "Slice: attribute 'ends' is mandatory.");
-  auto starts = getShape<ssize_t>(dict.at("starts"));
-  auto ends = getShape<ssize_t>(dict.at("ends"));
+  std::vector<ssize_t> starts;
+  std::vector<ssize_t> ends;
+  // Attribute 'axes' is optional.
+  std::vector<ssize_t> axes;
+  if (this->opsetVersion_ >= 10) {
+    Constant *startsC = getConstantByNameOrNull(op.input(1));
+    Constant *endsC = getConstantByNameOrNull(op.input(2));
+
+    RETURN_ERR_IF_NOT(startsC, "Starts Tensor is not Constant.");
+    RETURN_ERR_IF_NOT(endsC, "Ends Tensor is not Constant.");
+
+    if (startsC->getElementType() == ElemKind::Int64ITy) {
+      helperSetter<int64_t>(startsC, starts);
+    } else if (startsC->getElementType() == ElemKind::Int32ITy) {
+      helperSetter<int32_t>(startsC, starts);
+    } else {
+      RETURN_ERR_IF_NOT(false, "Starts Tensor has unsupported type.");
+    }
+
+    if (endsC->getElementType() == ElemKind::Int64ITy) {
+      helperSetter<int64_t>(endsC, ends);
+    } else if (endsC->getElementType() == ElemKind::Int32ITy) {
+      helperSetter<int32_t>(endsC, ends);
+    } else {
+      RETURN_ERR_IF_NOT(false, "Ends Tensor has unsupported type.");
+    }
+
+    if (op.input_size() > 3) {
+      Constant *axesC = getConstantByNameOrNull(op.input(3));
```

Comment:
Whoops just realized you also need to check if this is null and return an error, similar to above with `RETURN_ERR_IF_NOT(startsC, "Starts Tensor is not Constant.");`

Reply:
Good catch, thanks. I also added a check for Steps.